family_foto/utils/image.py

```diff
@@ -1,24 +1,43 @@
 import os
+import random
 
 from PIL import Image
+from cv2 import cv2
 from flask import current_app
 from resizeimage.resizeimage import resize_width
 
+from family_foto.logger import log
 
-def resize(path: str, filename: str, height: int, width: int):
+def resize(path: str, filename: str, height: int, width: int, force: bool = False):
     """
     Resizes an image to the ordered width,
     :param path: path to original image
     :param filename: name of the original image
     :param height: wanted height
     :param width: wanted width (aspect ration will be kept)
+    :param force: force recreation of thumbnail
     :return: path to resized image
     """
     with open(path, 'rb') as file:
         with Image.open(file) as image:
-            cover = resize_width(image, width)
             if not os.path.exists(current_app.config['RESIZED_DEST']):
                 os.mkdir(current_app.config['RESIZED_DEST'])
             save_path = f'{current_app.config["RESIZED_DEST"]}/{width}_{height}_{filename}'
+            if not force and os.path.exists(save_path):
+                log.warning(f'Thumbnail already exists: {save_path}')
+                return save_path
+            cover = resize_width(image, width)
```

Comment:
Please move this check up top

Reply:
done in: 812282e93a6ec9573c0d1266e53041be2ea92955